components/tooltip/base.ts

```diff
@@ -116,13 +113,24 @@ export abstract class NzTooltipBaseDirective implements OnChanges, OnDestroy, Af
 
   private delayTimer?: ReturnType<typeof setTimeout>;
 
-  constructor(
-    public elementRef: ElementRef,
-    protected hostView: ViewContainerRef,
-    protected renderer: Renderer2,
-    protected noAnimation?: NzNoAnimationDirective,
-    protected nzConfigService?: NzConfigService
-  ) {}
+  elementRef = inject(ElementRef);
+  protected hostView = inject(ViewContainerRef);
+  protected renderer = inject(Renderer2);
+  protected noAnimation = inject(NzNoAnimationDirective, { host: true, optional: true });
+  protected nzConfigService = inject(NzConfigService);
+  protected injector = inject(Injector);
+
+  constructor(protected componentType: Type<NzTooltipBaseComponent>) {}
+
+  ngOnInit(): void {
+    afterNextRender(
+      () => {
+        this.createComponent();
+        this.registerTriggers();
+      },
+      { injector: this.injector }
+    );
+  }
```

Comment:
Key: afterNextRender will only be executed in the client, which ensures that components are only dynamically created in the browser.